Solver/Dynamic/Ida.cpp

```diff
@@ -286,6 +286,11 @@ namespace Sundials
                 tout += dt;
             }
         }
+
+        model_->updateTime(tf, 0.0);
+        copyVec(yy_, model_->y());
+        copyVec(yp_, model_->yp());
+
```

Comment:
I am not sure copying vector data should be here. If it should, then other functions that drive simulations should be updated accordingly.